docs/concepts/models/model_kinds.md

```diff
@@ -261,5 +261,273 @@ FROM db.employees;
+
+Therefore you can use these models to not only tell you what the latest value is for a given record but also what the values were anytime in the past. Note that maintaining this history does come at a cost of increased storage and compute and this may not be a good fit for sources that change frequently since the history could get very large. 
+
+This example specifies a `SCD_TYPE_2` model kind:
+```sql linenums="1" hl_lines="3"
+MODEL (
+  name db.menu_items,
+  kind SCD_TYPE_2 (
+    unique_key (id),
+  )
+);
+
+SELECT
+  id::INT,
+  name::STRING,
+  price::DOUBLE,
+  updated_at::TIMESTAMP
+FROM 
+  stg.current_menu_items;
+```
+
+SQLMesh will materialize this table with the following structure:
+```sql linenums="1" hl_lines="3"
+TABLE db.menu_items (
+  id INT,
+  name STRING,
+  price DOUBLE,
+  updated_at TIMESTAMP,
+  valid_from TIMESTAMP,
+  valid_to TIMESTAMP
+);
+```
+
+### Column Names
+SQLMesh will automatically add the `valid_from` and `valid_to` columns to your table. If you would like to specify the names of these columns you can do so by adding the following to your model definition:
+```sql linenums="1" hl_lines="3"
+MODEL (
+  name db.menu_items,
+  kind SCD_TYPE_2 (
+    unique_key (id),
+    valid_from_name my_valid_from,
+    valid_to_name my_valid_to
+  )
+);
+```
+
+SQLMesh will materialize this table with the following structure:
+```sql linenums="1" hl_lines="3"
+TABLE db.menu_items (
+  id INT,
+  name STRING,
+  price DOUBLE,
+  updated_at TIMESTAMP,
+  my_valid_from TIMESTAMP,
+  my_valid_to TIMESTAMP
+);
+```
+
+The `updated_at` column name can also be changed by adding the following to your model definition:
+```sql linenums="1" hl_lines="3"
+MODEL (
+  name db.menu_items,
+  kind SCD_TYPE_2 (
+    unique_key (id),
+    updated_at_name my_updated_at
+  )
+);
+
+SELECT
+    id,
+    name,
+    price,
+    my_updated_at
+FROM
+    stg.current_menu_items;
+```
+
+SQLMesh will materialize this table with the following structure:
+```sql linenums="1" hl_lines="3"
+TABLE db.menu_items (
+  id INT,
+  name STRING,
+  price DOUBLE,
+  my_updated_at TIMESTAMP,
+  valid_from TIMESTAMP,
+  valid_to TIMESTAMP
+);
+```
+
+### Deletes
+
+Deletes are supported and when an item is detected as deleted then the `valid_to` column will be set to the time when SQLMesh started running (called `execution_time`). If a deleted column is added back in the source then it will be inserted back into the table with `valid_from` set to the largest of either the `updated_at` timestamp of the new record or the `valid_from` timestamp of the deleted record in the SCD Type 2 table.
+
+### Example of SCD Type 2 in Action
+
+Lets say that you started with the following data in your source table:
+
+| ID | Name             | Price |     Updated At      |
+|----|------------------|:-----:|:-------------------:|
+| 1  | Chicken Sandwich | 10.99 | 2020-01-01 00:00:00 |
```

Comment:
i'm afraid if we use time stamp strings like this we may have problems with time zones, i think different engines have different defaults

Reply:
Can you explain more your concern? This is just an example so obviously doesn't have an issue. Are you talking about where we use 1970 timestamp string when creating the actual scd type 2 query?

Reply:
we store the managed columns as timestamps right? for example if you store a timestamp in spark, it may use local time